pipeline/scripts/13-excelExport.R

```diff
@@ -264,143 +267,217 @@ IS_neg$Intensity <- as.numeric(as.character(IS_neg$Intensity))
+
+##########
+##### Line plots with all IS
+##########
+
+# function for ggplot theme
+# add smaller legend in the "all IS line plots", otherwise out-of-range when more than 13 IS lines
+theme_IS_line <- function(myPlot) {
+  myPlot + 
+    guides(
+      shape = guide_legend(override.aes = list(size = 0.5)),
+      color = guide_legend(override.aes = list(size = 0.5))) +
+    theme(
+      legend.title = element_text(size = 8), 
+      legend.text  = element_text(size = 6),
+      legend.key.size = unit(0.7, "line"),
+      axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8))
+}
 
-# Lineplot voor alle IS
-IS_neg_plot <- ggplot(IS_neg, aes(Sample,Intensity)) +
+# ggplot functions
+IS_neg_line_plot <- ggplot(IS_neg, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Neg)") +
   geom_point(aes(col=HMDB.name)) +
   geom_line(aes(col=HMDB.name, group=HMDB.name)) +
-  labs(x='',y='Intensity') +
-  theme(axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=8))
+  labs(x='',y='Intensity')
 
-IS_pos_plot <- ggplot(IS_pos, aes(Sample,Intensity)) +
+IS_pos_line_plot <- ggplot(IS_pos, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Pos)") +
   geom_point(aes(col = HMDB.name)) +
   geom_line(aes(col = HMDB.name, group = HMDB.name)) +
-  labs(x = '', y = 'Intensity') +
-  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8))
+  labs(x = '', y = 'Intensity')
 
-IS_sum_plot <- ggplot(IS_summed, aes(Sample, Intensity)) +
+IS_sum_line_plot <- ggplot(IS_summed, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Sum)") +
   geom_point(aes(col = HMDB.name)) +
   geom_line(aes(col = HMDB.name, group = HMDB.name)) +
-  labs(x = '', y = 'Intensity') +
-  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8))
-
-w <- 8 + 0.2 * len
-ggsave(paste0(outdir,"/plots/IS_line_all_neg.png"), plot = IS_neg_plot, height = w/2.5, width = w, units = "in")
-ggsave(paste0(outdir,"/plots/IS_line_all_pos.png"), plot = IS_pos_plot, height = w/2.5, width = w, units = "in")
-ggsave(paste0(outdir,"/plots/IS_line_all_sum.png"), plot = IS_sum_plot, height = w/2.5, width = w, units = "in")
-
-
-# Barplot voor selectie aan interne standaarden voor alle data
-IS_sum_selection <- c('2H8-Valine (IS)', '2H3-Leucine (IS)', '2H3-Glutamate (IS)', '2H4_13C5-Arginine (IS)', '13C6-Tyrosine (IS)')
-IS_pos_selection <- c('2H4-Alanine (IS)', '13C6-Phenylalanine (IS)', '2H4_13C5-Arginine (IS)', '2H3-Propionylcarnitine (IS)', '2H9-Isovalerylcarnitine (IS)')
+  labs(x = '', y = 'Intensity')
+
+# add theme to ggplot functions
+IS_sum_line_plot <- theme_IS_line(IS_sum_line_plot)
+IS_neg_line_plot <- theme_IS_line(IS_neg_line_plot)
+IS_pos_line_plot <- theme_IS_line(IS_pos_line_plot)
+
+# save plots to disk
+w <- 8 + 0.2 * sample_count
+ggsave(paste0(outdir,"/plots/IS_line_all_neg.png"), plot = IS_neg_line_plot, height = w/2.5, width = w, units = "in")
```

Comment:
See previous comment about height

Reply:
Also, see comment of ggsave function. I think you can make 1 function for Bar ánd Line plots

Reply:
see previous comment about the w, h and ggsave function